File: typescript/src/address/validate/validateMatchInfo.ts
import { RestApiAddressValidateMatchInfo } from "../../server/address/validate/restApiAddressValidateMatchInfo";
import { AddressAction, lookupAddressAction } from "./addressAction";
import { lookupPostalCodeAction, PostalCodeAction } from "./postalCodeAction";

export type ValidateMatchInfo = {
    postalCodeAction: PostalCodeAction;
    addressAction: AddressAction;
    generic_info?: string[];
    aus_info?: string[];
    fra_info?: string[];
    deu_info?: string[];
    gbr_info?: string[];
    nld_info?: string[];
    nzl_info?: string[];
    sgp_info?: string[];
};

export function restApiToValidateMatchInfo(response: RestApiAddressValidateMatchInfo): ValidateMatchInfo {
    return {
        postalCodeAction: lookupPostalCodeAction(response.postcode_action),
        addressAction: lookupAddressAction(response.address_action),
        generic_info: response.generic_info,
        aus_info: response.aus_info,
        deu_info: response.deu_info,
        fra_info: response.fra_info,
        gbr_info: response.gbr_info,
        nld_info: response.nld_info,
        nzl_info: response.nzl_info,
        sgp_info: response.sgp_info,
    }
}

File: csharp/DVSClient/DVSClient/address/validate/ValidateMatchInfo.cs
using DVSClient.Common;
using static DVSClient.Server.Address.Validate.RestApiAddressValidateResult;

namespace DVSClient.Address.Validate
{
    public class ValidateMatchInfo
    {
        public PostalCodeAction? PostalCodeAction { get; set; }

        public AddressAction? AddressAction { get; set; }

        public IEnumerable<string>? GenericInfo { get; set; }

        public IEnumerable<string>? AusInfo { get; set; }
        public IEnumerable<string>? DeuInfo { get; set; }
        public IEnumerable<string>? FraInfo { get; set; }
        public IEnumerable<string>? GbrInfo { get; set; }
        public IEnumerable<string>? NldInfo { get; set; }
        public IEnumerable<string>? NzlInfo { get; set; }
        public IEnumerable<string>? SgpInfo { get; set; }

        public ValidateMatchInfo(RestApiValidateMatchInfoFlags validateMatchInfo)
        {
            if (validateMatchInfo != null)
            {
                PostalCodeAction = validateMatchInfo.PostcodeAction?.GetEnumValueFromJsonName<PostalCodeAction>();
                AddressAction = validateMatchInfo.AddressAction?.GetEnumValueFromJsonName<AddressAction>();
                GenericInfo = validateMatchInfo.GenericInfo;
                AusInfo = validateMatchInfo.AusInfo;
                DeuInfo = validateMatchInfo.DeuInfo;
                FraInfo = validateMatchInfo.FraInfo;
                GbrInfo = validateMatchInfo.GbrInfo;
                NldInfo = validateMatchInfo.NldInfo;
                NzlInfo = validateMatchInfo.NzlInfo;
                SgpInfo = validateMatchInfo.SgpInfo;
            }
            else
            {
                PostalCodeAction = null;
                AddressAction = null;
                GenericInfo = null;
                AusInfo = null;
                DeuInfo = null;
                FraInfo = null;
                GbrInfo = null;
                NldInfo = null;
                NzlInfo = null;
                SgpInfo = null;
            }
        }
    }
}